### PR TITLE
BOAC-2353 Refresh authorized users from SIS advising tables

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2353

Tests to come in a follow-up; I wanted to give the job itself an early opportunity to get kicked around.

### CODE REVIEW

File: boac/models/authorized_user.py [hunk 1]
```diff
@@ -67,6 +67,15 @@ def __repr__(self):
                     created={self.created_at}>
                 """
 
+    @classmethod
+    def create_or_restore(cls, uid, is_admin=False, in_demo_mode=False):
+        existing_note = cls.query.filter_by(uid=uid).first()
```
Comment (johncrossman) [inline boac/models/authorized_user.py]: @paulkerschen Is `existing_note` a proper var name here?

Comment (paulkerschen) [reply boac/models/authorized_user.py]: @johncrossman It is not! Just pushed a follow-up commit.

File: boac/models/university_dept.py [hunk 2]
```diff
@@ -57,3 +60,22 @@ def create(cls, dept_code, dept_name, automate_memberships):
         db.session.add(dept)
         std_commit()
         return dept
+
+    def delete_all_members(self):
+        sql = """
+            DELETE FROM university_dept_members WHERE university_dept_id = :id;
+            UPDATE authorized_users SET deleted_at = now()
+                WHERE is_admin = false
```
Comment (johncrossman) [inline boac/models/university_dept.py]: @paulkerschen Not sure it matters: `IS false`  instead of  `= false`

**Comment by johncrossman** [comment]
Looks good @paulkerschen  Waiting on a happy Travis

**Pull request merged**